File: Beelina.APP/src/app/_models/report.ts
import { Entity } from './entity.model';

import { ReportControlsRelation } from './report-control-relation';
import { ReportCategoryEnum } from '../_enum/report-category.enum';

export class Report extends Entity {
  name: string;
  description: string;
  nameTextIdentifier: string;
  descriptionTextIdentifier: string;
  reportClass: string;
  custom: boolean;
  storedProcedureName: string;
  category: ReportCategoryEnum;
  reportControlsRelations: ReportControlsRelation[];

  constructor() {
    super();

    this.reportControlsRelations = new Array<ReportControlsRelation>();
  }
}

File: Beelina.LIB/Models/Report.cs
using Beelina.LIB.Enums;

namespace Beelina.LIB.Models
{
    public class Report
        : Entity
    {
        public string NameTextIdentifier { get; set; }
        public string DescriptionTextIdentifier { get; set; }
        public string ReportClass { get; set; }
        public bool Custom { get; set; }
        public bool Lock { get; set; }
        public string StoredProcedureName { get; set; }
        public ModulesEnum ModuleId { get; set; }
        public ReportCategoryEnum Category { get; set; }
        public PermissionLevelEnum UserMinimumModulePermission { get; set; }
        public PermissionLevelEnum UserMaximumModulePermission { get; set; }
        public string OnlyAvailableOnBusinessModel { get; set; }

        public List<ReportControlsRelation> ReportControlsRelations { get; set; }
    }
}